### PR TITLE
Add Restart File Support for Spiral Inflow-Control Devices

### PR DESCRIPTION
**Pull request by bska**
This commit identifies well segments that correspond to SICDs (input keyword `WSEGSICD`) and captures those in the ISEG vector.  Furthermore, we output characteristic properties of such SICD segments (e.g, the length, the base strength, calibration fluid properties) in the RSEG vector.

Thanks to @tskille and @jalvestad for invaluable assistance in identifying these items.

---

Note that this PR depends on both PRs #1269 and #1251 and must not be merged before those are installed in master. On top of those PRs, we change only the `AggregateMSWData.cpp` file. This PR can be used separately for testing purposes.

### CODE REVIEW

**Comment by bska** [comment]
Marking this as ready for review now that the prerequisite PRs (#1251 and #1261) have been merged into master.

File: src/opm/output/eclipse/AggregateMSWData.cpp [hunk 1]
```diff
@@ -459,6 +477,48 @@ namespace {
...
+                                            RSegArray&               rSeg)
+        {
+            using Ix = ::Opm::RestartIO::Helpers::VectorItems::RSeg::index;
+            using M  = ::Opm::UnitSystem::measure;
+
+            const auto& sicd = segment.spiralICD();
+
+            // Convert to [Pressure]/([Rate] * [Rate)]--Rate at reservoir cond.
+            rSeg[baseIndex + Ix::DeviceBaseStrength] = usys.from_si(M::pressure,
+                usys.to_si(M::rate, usys.to_si(M::rate, sicd->strength())));
```
Comment (joakim-hove) [inline src/opm/output/eclipse/AggregateMSWData.cpp]: I would have guessed that the innermost conversion should have been: `usys.from_si(M::rate, sicd->string())` - now it looks like the `sicd->strength()` is in deck units?

**Comment by joakim-hove** [comment]
One question about a unit conversion which looked suspicious to me - otherwise good to go.

File: src/opm/output/eclipse/AggregateMSWData.cpp (hunk 1 above)
Comment (bska) [reply src/opm/output/eclipse/AggregateMSWData.cpp]: I **believe** it is correct to use `to_si()` here, although I did have think long and hard about this.  It is decidedly subtle.  Feel free to check my logic.

We're trying to create a conversion from internal (i.e., strictly SI) units, but to an output unit that is a ratio of the form
```
     [Pressure]
 -----------------
  [Rate] * [Rate]
```
Since the `[Rate]` is in the denominator we need to divide (twice) with the conversion factor (SI Rate -> Output Rate).  Assume "METRIC" conventions for simplicity.  Then SI Rate -> Output Rate is the same as (1 / SI Time) -> (1 / Metric Time) which is (Metric Time) -> (SI Time) = multiply by 86,400.

In other words we somehow need to manufacture a conversion constant of
```
      1.0e-5
--------------------
  86,400 * 86,400
```
between the `sicd->strength()` in internal units&mdash; in general very large numbers in the order of `1.0e+14`&mdash;and the output item in deck units.  On the other hand, when implementing conversions that don't already exist in the `UnitSystem::measure` by chaining together conversion calls we only really get to use multiplication as a basic operation.  That's why `.to_si()` effectively plays the role of "`* 1 / .from_si()`".

Does that make sense?

Comment (joakim-hove) [reply src/opm/output/eclipse/AggregateMSWData.cpp]: > Does that make sense?

yes it makes sense; seems to correct to me. But maybe  a better solution would be to actually add the dimension `Pressure / Rate^2` to the unit system?

**Comment by bska** [comment]
I have rebased this on top of #1307 in order to use the dedicated units introduced there.  Hopefully the unit conversion in `assignSpiralICDCharacteristics()` are easier to read now.

**Comment by bska** [comment]
The prerequisite PR #1307 has been merged into master.  I'm running a build test here.  I expect [`TEST_WSEGSICD`](https://github.com/OPM/opm-tests/blob/6c406ed8f81e3005d8607951947a9bcb15dbc8fb/wsegsicd/TEST_WSEGSICD.DATA) to fail since this PR will be outputting additional data items to the restart file, but no other regression failures are expected.  Any such other failures are bugs in this PR.

**Comment by bska** [comment]
jenkins build this please

**Comment by GitPaean** [comment]
jenkins build this update_data please

**Comment by GitPaean** [comment]
jenkins build this opm-tests=208 please

**Comment by bska** [comment]
jenkins build this opm-tests=208 please

**Comment by bska** [comment]
I don't have anything else to add here.  May I solicit a review please?

**Comment by bska** [comment]
jenkins build this please

**Comment by bska** [comment]
jenkins build this opm-tests=208 please

**Comment by bska** [comment]
Associate data update has propagated to the CI system.  I'm merging this now per prior approval.
